### PR TITLE
Add xsettingsd package

### PR DESCRIPTION
**Pull request by clefru**
First contribution. Please let me know if there's something to improve.

### CODE REVIEW

File: pkgs/tools/X11/xsettingsd/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    sha256 = "1yz9hkjgyyp3yjjk0x1j44zbgm3m5kn5xqylip533vxvfg06qnjr";
+  };
+
+  patches = [
+    ./SConstruct.patch
+  ];
+
+  buildInputs = [ libX11 scons pkgconfig ];
+  buildPhase = ''
+    runHook preBuild
```
Comment (lethalman) [inline pkgs/tools/X11/xsettingsd/default.nix]: Doesn't make much sense to `runHook preBuild` here, you can remove it.

File: pkgs/tools/X11/xsettingsd/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    mkdir -p "$out"
+    scons \
+      -j$NIX_BUILD_CORES -l$NIX_BUILD_CORES \
+      "prefix=$out"
+    runHook postBuild
+  '';
+  
+  installPhase = ''
+    runHook preInstall
```
Comment (lethalman) [inline pkgs/tools/X11/xsettingsd/default.nix]: Same here.

File: pkgs/tools/X11/xsettingsd/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    mkdir -p "$out"/bin
+    install xsettingsd "$out"/bin
+    install dump_xsettings "$out"/bin
+    runHook postInstall
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Provides settings to X11 applications via the XSETTINGS specification.";
```
Comment (lethalman) [inline pkgs/tools/X11/xsettingsd/default.nix]: A nitpick, can you please remove that trailing dot in the description? :)

**Comment by lethalman** [comment]
Welcome. I've commented a couple of nitpicks just to line out some of nixpkgs conventions, otherwise the package looks quite good :)

File: pkgs/tools/X11/xsettingsd/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{ stdenv, fetchgit, scons, libX11, pkgconfig }:
+
+stdenv.mkDerivation rec {
+  name = "xsettingsd-${version}";
+  version = "0.0.1";
+
+  src = fetchgit {
```
Comment (jagajaga) [inline pkgs/tools/X11/xsettingsd/default.nix]: Better use `fetchFromGithub`

**Comment by clefru** [comment]
Thanks for the input. I addressed all remarks.

My package builds successfully locally, however Travis seems to not like it. From its logs it doesn't seems that my package is the culprit but rather something internal with Travis isn't right. Should I be worried?

**Comment by lethalman** [comment]
Built locally, thanks.

**Pull request merged**